File: core/domain/src/main/java/com/gaugustini/mhfudatabase/domain/model/Skill.kt
package com.gaugustini.mhfudatabase.domain.model

/**
 * Represents a skill that can be activated by accumulating skill points.
 *
 * @property id The unique identifier of the skill.
 * @property skillTreeId The ID of the skill tree this skill belongs to.
 * @property name The name of the skill.
 * @property description A description of the skill's effects.
 * @property requiredPoints The number of points required to activate this skill.
 */
data class Skill(
    val id: Int,
    val skillTreeId: Int,
    val name: String,
    val description: String,
    val requiredPoints: Int,
)

File: lib/features/skill/domain/skill.dart
import '../../../core/domain/enums.dart';

class const SkillTree({
  required final int id,
  required final String name,
  required final SkillCategory category,
  final List<Skill>? skills,
});

class const Skill({
  required final int id,
  required final int skillTreeId,
  required final String name,
  required final String description,
  required final int requiredPoints,
});
